openlibrary/catalog/marc/parse.py

```diff
@@ -326,51 +321,65 @@ def read_languages(rec, lang_008: Optional[str] = None):
     return [lang_map.get(code, code) for code in found]
 
 
-def read_pub_date(rec):
+def read_pub_date(rec: MarcBase) -> str | None:
     fields = rec.get_fields('260')
-    if not fields:
-        return
     found = []
     for f in fields:
-        found += [v for v in f.get_subfield_values('c') if v]
+        found += f.get_subfield_values('c')
     return remove_trailing_number_dot(found[0].strip('[]')) if found else None
 
 
-def read_publisher(rec):
-    fields = rec.get_fields('260') or rec.get_fields('264')[:1]
+def read_publisher(rec: MarcBase) -> dict[str, Any] | None:
+    fields = (
+        rec.get_fields('260')
+        or rec.get_fields('264')[:1]
+        or [rec.get_linkage('260', '880')]
+    )
     if not fields:
-        return
+        return None
     publisher = []
     publish_places = []
     for f in fields:
-        f.remove_brackets()
-        contents = f.get_contents(['a', 'b'])
+        contents = f.get_contents('ab')
         if 'b' in contents:
-            publisher += [x.strip(" /,;:") for x in contents['b']]
+            publisher += [x.strip(" /,;:[") for x in contents['b']]
         if 'a' in contents:
-            publish_places += [x.strip(" /.,;:") for x in contents['a'] if x]
+            publish_places += [x.strip(" /.,;:[") for x in contents['a']]
     edition = {}
     if publisher:
-        edition["publishers"] = publisher
+        edition['publishers'] = publisher
     if len(publish_places) and publish_places[0]:
-        edition["publish_places"] = publish_places
+        edition['publish_places'] = publish_places
     return edition
 
 
-def read_author_person(f):
-    f.remove_brackets()
+def name_from_list(name_parts: list[str]) -> str:
+    STRIP_CHARS = r' /,;:[]'
+    name = ' '.join(strip_foc(s).strip(STRIP_CHARS) for s in name_parts)
+    return remove_trailing_dot(name)
+
+
+def read_author_person(field: MarcFieldBase, tag: str = '100') -> dict | None:
+    """
+    This take either a MARC 100 Main Entry - Personal Name (non-repeatable) field
+      or
+    700 Added Entry - Personal Name (repeatable)
+      or
+    720 Added Entry - Uncontrolled Name (repeatable)
+    and returns an author import dict.
+    """
     author = {}
-    contents = f.get_contents(['a', 'b', 'c', 'd', 'e'])
+    contents = field.get_contents('abcde6')
```

Comment:
$0 and $1 contain strong identifiers which could greatly simplify author matching (for 100 & 700, not 720)

Reply:
Raised as a new feature: #7724 I agree, these look promising. We should at least import them, and ideally make use of them in the author matching.